src/EventListener/ReadListener.php

```diff
@@ -28,11 +30,17 @@
 {
     private $collectionDataProvider;
     private $itemDataProvider;
+    private $subresourceDataProvider;
 
-    public function __construct(CollectionDataProviderInterface $collectionDataProvider, ItemDataProviderInterface $itemDataProvider)
+    public function __construct(CollectionDataProviderInterface $collectionDataProvider, ItemDataProviderInterface $itemDataProvider, SubresourceDataProviderInterface $subresourceDataProvider = null)
     {
         $this->collectionDataProvider = $collectionDataProvider;
         $this->itemDataProvider = $itemDataProvider;
+        $this->subresourceDataProvider = $subresourceDataProvider;
+
+        if (null === $subresourceDataProvider) {
```

Comment:
Same here, I would just disable this feature if the `subresourceDataProvider` is not passed, without triggering a deprecation notice.